simulator/front_end/src/front_end/region_creation/fixed_region.py

```diff
@@ -28,29 +16,28 @@ def create_fixed_regions(hash_file: HashFile, size_mib: int) -> Generator[Region
     :return:
     """
     max_region_size = (size_mib * 1024 * 1024)
-    current_region = Region(max_region_size)
+    region = Region(max_region_size)
 
     while hash_file.num_files_processed < hash_file.total_files:
         hash_file.hashfile_next_file()
         while hash_file.num_hashes_processed_current_file < hash_file.current_file_total_chunks:
-            fingerprint, compression = hash_file.hashfile_next_chunk()
-            if len(fingerprint) + current_region.current_size > max_region_size:
-                yield current_region
-                current_region = Region(max_region_size)
+            fingerprint, compression, current_chunk_size = hash_file.hashfile_next_chunk()
+            if current_chunk_size + region.current_size > max_region_size:
```

Comment:
I think current_chunk_size is the size of the actual data. But the fingerprint hashes will all be a uniform size. The fingerprints are what we are filling the regions up with. So I think we need to count how many bytes the fingerprint itself has.

Reply:
Nevermind, you are right